trieste/models/optimizer.py

```diff
@@ -19,8 +19,10 @@
 from typing import Any, Callable, Dict, Iterable, Optional, Tuple, Union
 
 import gpflow
+from gpflux.models import DeepGP
```

Comment:
right, we should probably have `optimizer.py` file in `models/gpflux/` with gpflux specific stuff

optimizers should have been decomposed alongside the model_interface decomposition but to me it seemed better to leave it for a separate PR

perhaps you can try moving this new code to gpflux folder?

Reply:
I've had a go at moving the model-specific parts to `optimizer.py` files in the `gpflux` and `gpflow` folders